File: jupyter2/ps/typedef/time/jl/time.jl
### begin my answer

### begin hidden
struct Time
    hour::Int
    minute::Int
    second::Float64
end

second_of_time(t::Time) = t.hour * 3600 + t.minute * 60 + t.second

function time_diff(a::Time, b::Time)
    abs(second_of_time(a) - second_of_time(b))
end
### end hidden
### end my answer

function main()
    @assert abs(time_diff(Time(12, 0, 0.), Time(11, 30, 0.)) - 1800.) < 1e-6
    @assert abs(time_diff(Time(23, 59, 59.9), Time(0, 0, 0.)) - 86399.9) < 1e-6
    println("OK")
end

main()

File: jupyter2/ps/typedef/time/rs/time.rs
/** begin my answer */

/** begin hidden */
struct Time {
	hour: u8,
	minute: u8,
	second: f64,
}

fn second_of_time(t: Time) -> f64 {
	3600.0 * (t.hour as f64) + 60.0 * (t.minute as f64) + t.second
}

fn time_diff(a : Time, b : Time) -> f64 {
    second_of_time(a) - second_of_time(b)
}
/** end hidden */
/** end my answer */

fn main() {
    assert!((time_diff(Time {hour: 12, minute: 0, second: 0.0}, Time {hour: 11, minute: 30, second: 0.0}) - 1800.0).abs() < 1e-6);
    assert!((time_diff(Time {hour: 23, minute: 59, second: 59.9}, Time {hour: 0, minute: 0, second: 0.0}) - 86399.9).abs() < 1e-6);
    println!("OK")
}
